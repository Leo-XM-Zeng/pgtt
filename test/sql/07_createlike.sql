----
-- Regression test to Global Temporary Table implementation
--
-- Test for GTT with TABLE ... (LIKE ...) clause.
--
----

-- Import the library
LOAD 'pgtt';

-- Create a GTT like table to test ON COMMIT PRESERVE ROWS
CREATE GLOBAL TEMPORARY TABLE t_glob_temptable1 (
	LIKE source
	INCLUDING ALL
) ON COMMIT PRESERVE ROWS;

-- Look at table description
\d+ t_glob_temptable1

-- Look at Global Temporary Table definition
SELECT nspname, relname, preserved, code FROM pgtt_schema.pg_global_temp_tables;

-- A "template" unlogged table should exists
SELECT n.nspname, c.relname FROM pg_class c JOIN pg_namespace n ON (c.relnamespace=n.oid) WHERE relname = 't_glob_temptable1';

BEGIN;

-- With the first insert some value in the temporary table
INSERT INTO t_glob_temptable1 VALUES (1, 'One');

-- Look at temp table description
\d+ t_glob_temptable1

-- Look if we have two tables now
SELECT n.nspname, c.relname FROM pg_class c JOIN pg_namespace n ON (c.relnamespace=n.oid) WHERE relname = 't_glob_temptable1';

-- Look at content of the template for Global Temporary Table, must be empty
SET pgtt.enabled TO off;
SELECT * FROM pgtt_schema.t_glob_temptable1;
SET pgtt.enabled TO on;

-- Look at content of the Global Temporary Table
SELECT * FROM t_glob_temptable1 ORDER BY id;
COMMIT;

SELECT * FROM t_glob_temptable1 ORDER BY id;

-- Reconnect and drop it
\c - -
LOAD 'pgtt';

-- Cleanup
DROP TABLE t_glob_temptable1;

