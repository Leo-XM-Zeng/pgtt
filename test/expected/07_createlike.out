----
-- Regression test to Global Temporary Table implementation
--
-- Test for GTT with TABLE ... (LIKE ...) clause.
--
----
-- Import the library
LOAD 'pgtt';
-- Create a GTT like table to test ON COMMIT PRESERVE ROWS
CREATE GLOBAL TEMPORARY TABLE t_glob_temptable1 (
	LIKE source
	INCLUDING ALL
) ON COMMIT PRESERVE ROWS;
WARNING:  GLOBAL is deprecated in temporary table creation
LINE 1: CREATE GLOBAL TEMPORARY TABLE t_glob_temptable1 (
               ^
-- Look at table description
\d+ t_glob_temptable1
                                                Unlogged table "pgtt_schema.t_glob_temptable1"
 Column |         Type          | Collation | Nullable |              Default               | Storage  | Stats target |      Description      
--------+-----------------------+-----------+----------+------------------------------------+----------+--------------+-----------------------
 id     | integer               |           | not null | nextval('source_id_seq'::regclass) | plain    |              | auto generated column
 c2     | character varying(50) |           | not null |                                    | extended |              | 
 lbl    | character varying     |           |          | '-'::character varying             | extended |              | 
Indexes:
    "t_glob_temptable1_pkey" PRIMARY KEY, btree (id)
    "t_glob_temptable1_c2_key" UNIQUE CONSTRAINT, btree (c2)
    "t_glob_temptable1_lbl_idx" btree (lbl)

-- Look at Global Temporary Table definition
SELECT nspname, relname, preserved, code FROM pgtt_schema.pg_global_temp_tables;
   nspname   |      relname      | preserved |         code          
-------------+-------------------+-----------+-----------------------
 pgtt_schema | t_glob_temptable1 | t         |                      +
             |                   |           |         LIKE source  +
             |                   |           |         INCLUDING ALL+
             |                   |           | 
(1 row)

-- A "template" unlogged table should exists
SELECT n.nspname, c.relname FROM pg_class c JOIN pg_namespace n ON (c.relnamespace=n.oid) WHERE relname = 't_glob_temptable1';
   nspname   |      relname      
-------------+-------------------
 pgtt_schema | t_glob_temptable1
(1 row)

BEGIN;
-- With the first insert some value in the temporary table
INSERT INTO t_glob_temptable1 VALUES (1, 'One');
-- Look at temp table description
\d+ t_glob_temptable1
                                                     Table "pg_temp_3.t_glob_temptable1"
 Column |         Type          | Collation | Nullable |              Default               | Storage  | Stats target |      Description      
--------+-----------------------+-----------+----------+------------------------------------+----------+--------------+-----------------------
 id     | integer               |           | not null | nextval('source_id_seq'::regclass) | plain    |              | auto generated column
 c2     | character varying(50) |           | not null |                                    | extended |              | 
 lbl    | character varying     |           |          | '-'::character varying             | extended |              | 
Indexes:
    "t_glob_temptable1_pkey" PRIMARY KEY, btree (id)
    "t_glob_temptable1_c2_key" UNIQUE CONSTRAINT, btree (c2)
    "t_glob_temptable1_lbl_idx" btree (lbl)

-- Look if we have two tables now
SELECT n.nspname, c.relname FROM pg_class c JOIN pg_namespace n ON (c.relnamespace=n.oid) WHERE relname = 't_glob_temptable1';
   nspname   |      relname      
-------------+-------------------
 pgtt_schema | t_glob_temptable1
 pg_temp_3   | t_glob_temptable1
(2 rows)

-- Look at content of the template for Global Temporary Table, must be empty
SET pgtt.enabled TO off;
SELECT * FROM pgtt_schema.t_glob_temptable1;
 id | c2 | lbl 
----+----+-----
(0 rows)

SET pgtt.enabled TO on;
-- Look at content of the Global Temporary Table
SELECT * FROM t_glob_temptable1 ORDER BY id;
 id | c2  | lbl 
----+-----+-----
  1 | One | -
(1 row)

COMMIT;
SELECT * FROM t_glob_temptable1 ORDER BY id;
 id | c2  | lbl 
----+-----+-----
  1 | One | -
(1 row)

-- Reconnect and drop it
\c - -
LOAD 'pgtt';
-- Cleanup
DROP TABLE t_glob_temptable1;
